pyproject.toml

```
@@ -1,20 +1,184 @@
-  "numpy",
-  "scipy",
-  "qiskit",
-  "opt_einsum",
-  "tqdm"
+maintainers = [
+  { name="Lukas Burgholzer", email = "burgholzer@me.com" },
 ]
 readme = "README.md"
-requires-python = ">=3.8"
+requires-python = ">=3.9"
+dynamic = ["version"]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "Topic :: Scientific/Engineering :: Electronic Design Automation (EDA)",
+    "Typing :: Typed",
+]
+keywords = ["MQT", "quantum-computing", "quantum-simulation", "tensor-networks"]
+
+dependencies = [
+    "numpy>=2.1; python_version >= '3.13'",
+    "numpy>=1.26; python_version >= '3.12'",
+    "numpy>=1.24; python_version >= '3.11'",
+    "numpy>=1.22.4",
+    "scipy>=1.13.1",
+    "scipy>=1.14.1; python_version >= '3.13'",
+    "qiskit>=1.0.0",
+    "qiskit>=1.3.0; python_version >= '3.13'",
+    "opt-einsum>=3.4.0",
+    "tqdm>=4.67.1",
+]
+
+[project.urls]
+Homepage = "https://github.com/cda-tum/mqt-yaqs"
+Documentation = "https://mqt.readthedocs.io/projects/yaqs"
+Issues = "https://github.com/cda-tum/mqt-yaqs/issues"
+Discussions = "https://github.com/cda-tum/mqt-yaqs/discussions"
+
+
+[tool.hatch.version]
+source = "vcs"
+
+[tool.hatch.build.targets.wheel]
+packages = ["src/mqt"]
+
+[tool.hatch.build.hooks.vcs]
+version-file = "src/mqt/yaqs/_version.py"
+
+
+[tool.pytest.ini_options]
+minversion = "7.2"
+addopts = ["-ra", "--strict-markers", "--strict-config"]
+xfail_strict = true
+filterwarnings = [
+    "error",
+    # Qiskit 1.3 deprecations
+    'ignore:.*``qiskit.dagcircuit.dagcircuit.DAGCircuit.*`` is deprecated as of qiskit 1.3.0.*:DeprecationWarning:',
+]
+log_cli_level = "info"
+testpaths = ["tests"]
+
+
+[tool.coverage]
+run.source = ["mqt.yaqs"]
+report.exclude_also = [
+    '\.\.\.',
+    'if TYPE_CHECKING:',
+    'raise AssertionError',
+    'raise NotImplementedError',
+    'def __dir__()',  # Ignore __dir__ method that exists mainly for better IDE support
+    '@overload'  # Overloads are only for static typing
+]
+
+
+[tool.mypy]
+files = ["src/mqt", "tests", "examples", "noxfile.py"]
+mypy_path = ["$MYPY_CONFIG_FILE_DIR/src"]
+python_version = "3.9"
+warn_unused_configs = true
+enable_error_code = ["ignore-without-code", "redundant-expr", "truthy-bool"]
+strict = true
+disallow_untyped_defs = false
+explicit_package_bases = true
+warn_unreachable = true
+
+[[tool.mypy.overrides]]
+module = ["qiskit.*", "matplotlib.*", "qiskit_aer.*", "opt_einsum.*"]
```

```suggestion
module = ["qiskit.*", "matplotlib.*", "qiskit_aer.*", "opt_einsum.*", "qutip.*]
```
qutip does not contain a `py.typed` file and is thus not marked as typed. Including it in the list here should make the mypy warnings go away.

---

I removed qutip entirely from the example since it was just used as comparison. Should we leave this still?

---

If qutip was removed, this suggestion does no longer really apply and can be resolved 🙂